part_2/second_split/method_v2.R

```diff
@@ -0,0 +1,398 @@
+rm(list = ls())
+setwd("/Volumes/GoogleDrive/My Drive/alsta_analysis_LA/second_split")
+
+######################################
+##### Section 1: Import packages #####
+######################################
+library(forecast)
+library(bsts)
+library(glmnet)
+library(plyr)
+library(lme4)
+library(strucchange)
+library(changepoint)
+library(mice)
+source("functions_v2.R")
+
+####################################
+##### Section 2: Preprocessing #####
+####################################
+
+### Impute data ###
+# Remove future data.
+end_date <- as.POSIXlt("2018-06-21", tz = "America/Los_Angeles")
```

Comment:
I would recommend moving to dates rather than timestamps. It will be easier to manage and doesn't require tracking of the timezone.

Reply:
Changed.